src/index.ts

```diff
@@ -1 +1,326 @@
+ */
+export const getAndroidDevToolPath = async (tool: AndroidTool): Promise<string> => {
+    const { androidHome } = await ensureSdkmanager();
+
+    if (typeof tool === 'string') {
+        if (tool in unversionedAndroidTools) {
+            const relativeToolPath = unversionedAndroidTools[tool as keyof typeof unversionedAndroidTools].path;
+            const toolPath = join(androidHome, relativeToolPath);
+            if (!(await fs.exists(toolPath))) return await installAndroidDevTool(tool);
+            return toolPath;
+        } else if (tool in versionedAndroidTools) {
+            const relativeGlobToolPath = versionedAndroidTools[tool as keyof typeof versionedAndroidTools].path('*');
+            const relativeRegexToolPath =
+                versionedAndroidTools[tool as keyof typeof versionedAndroidTools].path('(.+)');
+
+            const installedTools = await globby(relativeGlobToolPath, { cwd: androidHome });
+            const installedVersions = installedTools
+                .map((p) => p.match(relativeRegexToolPath)?.[1])
+                .filter((v): v is string => !!v);
+            const latestVersion = getLatestVersion(installedVersions);
+
+            if (!latestVersion) return await installAndroidDevTool(tool);
+
+            return join(
+                androidHome,
+                versionedAndroidTools[tool as keyof typeof versionedAndroidTools].path(latestVersion)
+            );
+        }
+
+        throw new Error(`Unsupported tool: "${tool}"`);
+    }
+
+    const { tool: toolName, packageVersion } = tool;
+    if (!(toolName in versionedAndroidTools)) throw new Error(`Unsupported: ${toolName}`);
+
+    const relativeToolPath = versionedAndroidTools[toolName].path(packageVersion);
+    const toolPath = join(androidHome, relativeToolPath);
+
+    if (!(await fs.exists(toolPath))) return await installAndroidDevTool(tool);
+    return toolPath;
+};
+
+/**
+ * Run an Android development tool. If the tool is not installed yet, it will automatically be installed. The tool is
+ * executed using [`execa`](https://github.com/sindresorhus/execa).
+ *
+ * If no version is specified for a versioned tool, the latest installed (if the tool was already installed) or
+ * available version (if it isn't installed yet) is used.
+ *
+ * @param tool The tool to run, either as just its name or with an explicit version.
+ * @param args The arguments to pass to the command.
+ * @param execaOptions The options to pass to execa, see: https://github.com/sindresorhus/execa#options-1.
+ *
+ * @returns The result from execa, see: https://github.com/sindresorhus/execa#childprocess.
+ */
+export const runAndroidDevTool = async (tool: AndroidTool, args?: string[], execaOptions?: ExecaOptions) => {
+    const { androidHome } = await ensureSdkmanager();
+
+    const toolPath = await getAndroidDevToolPath(tool);
+    return execa(toolPath, args, { ...execaOptions, env: { ANDROID_HOME: androidHome, ...execaOptions?.env } });
```

Comment:
Maybe it would be nice to add the `ANDROID_HOME` to the `PATH` as well. in case this runs scripts which would try to call those?

Reply:
There isn't anything in `$ANDROID_HOME` that can be executed. You'd have to add the individual binary directories. And those heavily depend on which tools and which versions of those are installed. And if you have multiple versions of a package, which should we add to the `$PATH` (/in which order)?

My initial idea was to just provide a `$PATH` that the user can use, but I really don't think that is feasible given how the Android SDK works. That's why I went with this approach of calling the binaries directly.  
And according to https://developer.android.com/tools/variables, you only need to set the `$PATH` to "run tools from the command line without including the full path to the executable".